pkg/ingester/flush.go

```diff
@@ -179,13 +183,13 @@ func (i *Ingester) shouldFlushSeries(series *memorySeries, fp model.Fingerprint,
 		return reasonMultipleChunksInSeries
 	} else if len(series.chunkDescs) > 0 {
 		// Otherwise look in more detail at the first chunk
-		return i.shouldFlushChunk(series.chunkDescs[0], fp)
+		return i.shouldFlushChunk(series.chunkDescs[0], fp, value.IsStaleNaN(float64(series.lastSampleValue)))
```

Comment:
Should you also check `series.lastSampleValueSet`?

_Same comment applies below in `flushUserSeries()`_

Reply:
Also, may be more readable if you add a simple function `series.stale()` which returns a `bool`.